src/v/kafka/server/handlers/handler_probe.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "kafka/server/handlers/handler_probe.h"
+
+#include "config/configuration.h"
+#include "kafka/server/handlers/handler_interface.h"
+#include "prometheus/prometheus_sanitize.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/metrics.hh>
+
+#include <chrono>
+#include <memory>
+
+namespace kafka {
+
+handler_probe_manager::handler_probe_manager()
+  : _probes(max_api_key() + 2)
+  , _unknown_handler_index(max_api_key() + 1) {
+    for (size_t i = 0; i < _probes.size(); i++) {
+        auto key = api_key{i};
+
+        if (handler_for_key(key) || i == _unknown_handler_index) {
+            _probes[i] = std::make_unique<handler_probe>(key);
+            _probes.at(i)->setup_metrics();
```

Comment:
Any reason why `at()` here vs `operator[]` above?

Reply:
Probably just some bad habits. `operator[]` can cause insertion in the map types so I'm just to avoiding it for index-only operations. Will switch to using `operator[]` here though.

Reply:
Switched all calls from `at` to `operator[]`